### PR TITLE
GetAppmenus: ensure right app directories

### PR DESCRIPTION
**Pull request by pwmarcz**
The script depends on XDG_DATA_DIRS environment variable
being set up correctly, which is not the case when it is
running under sudo. As a result, a post-install trigger
for apt could remove application entries from other sources
(Snap, Flatpak).

Fixes QubesOS/qubes-issues#5477.

### CODE REVIEW

**Comment by marmarek** [comment]
ShellCheck hates you (see failed travis job), please add a comment to ignore this warning (`shellcheck disable=SC1090`).

File: qubes-rpc/qubes.GetAppmenus [hunk 1]
```diff
@@ -6,6 +6,16 @@
 #  - directories stored on /rw in case of "rw-only" persistence
 #  - nothing, otherwise
 
+# Reload scripts in /etc/profile.d/, in case they register additional
+# directories in XDG_DATA_DIRS and we forgot them
+# (e.g. because we are running under sudo).
+for i in /etc/profile.d/*.sh ; do
+    if [ -r "$i" ]; then
+	# shellcheck disable=SC1090
```
Comment (marmarek) [inline qubes-rpc/qubes.GetAppmenus]: Indentation is odd (3 instead of 4 spaces).

Comment (pwmarcz) [reply qubes-rpc/qubes.GetAppmenus]: Ouch, that's a tab, my editor misbehaved. I pushed an updated commit.

**Comment by marmarek** [comment]
While at it, I think there is very similar issue with `~/.local/share/applications`. The script use `$HOME`, but if that's `/root`, then entries in `/home/user/.local/share/applications` will be lost. As you know from the other issue, hardcoding `/home/user` is a bad idea, but maybe we can use "default user" if the script is running as root?

**Comment by pwmarcz** [comment]
How about this?

**Comment by marmarek** [comment]
Lets see what shellcheck says, but should be fine.

**Pull request merged**